core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -189,8 +194,12 @@ public static LoadTableResponse stageTableCreate(Catalog catalog, Namespace name
         location,
         properties);
 
+    TableMetadata metadataWithLocation = TableMetadata.buildFrom(metadata)
+        .withMetadataLocation(metadataLocation)
+        .build();
+
     return LoadTableResponse.builder()
-        .withTableMetadata(metadata)
+        .withTableMetadata(metadataWithLocation)
```

Comment:
This change causes the number of tests in `TestRESTCatalog` that fail if we require `metadataLocation` be non-null via the builder to go from 16 to 13.

If we remove that requirement from the builder, then all tests pass. Given that we use `LoadTableResponse` in various places, I think we should drop the requirement that `metadataLocation` be present at all times and allow for it to be `null`.